Getting "Error: Cannot find module 'next/headers'" when accessing "api/auth/signin"

### Question 💬

Hello I am trying to implement Facebook Login but I am getting this error:

`error - Error: Cannot find module 'next/headers'
Require stack:
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/index.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/.next/server/pages/api/auth/[...nextauth].js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next-server.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/lib/start-server.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/cli/next-dev.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/lib/commands.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js:15:16)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/index.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/.next/server/pages/api/auth/[...nextauth].js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next-server.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/lib/start-server.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/cli/next-dev.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/lib/commands.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/bin/next'
  ]
}`



### How to reproduce ☕️

Here is the code if you want to try by yourself https://github.com/edgarlcs/rifadisimo

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

--- a/.github/version-pr/action.yml
+++ b/.github/version-pr/action.yml
@@ -4,5 +4,5 @@ outputs:
   version:
     description: "npm package version"
 runs:
-  using: "node18"
+  using: "node16"
   main: "index.js"

--- a/packages/next-auth/src/next/index.ts
+++ b/packages/next-auth/src/next/index.ts
@@ -1,7 +1,6 @@
 import { NextAuthHandler } from "../core"
 import { detectHost } from "../utils/detect-host"
 import { setCookie } from "./utils"
-import { cookies as nextCookies, headers } from "next/headers"
 
 import type {
   GetServerSidePropsContext,
@@ -121,18 +120,18 @@ export async function unstable_getServerSession(
     experimentalRSCWarningShown = true
   }
 
-  const [req, res, options] = isRSC
-    ? [
-        {
-          headers: headers(),
-          cookies: nextCookies()
-            .getAll()
-            .reduce((acc, c) => ({ ...acc, [c.name]: c.value }), {}),
-        } as any,
-        { getHeader() {}, setCookie() {}, setHeader() {} } as any,
-        args[0],
-      ]
-    : args
+  let req, res, options: NextAuthOptions
+  if (isRSC) {
+    options = args[0]
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { headers, cookies } = require("next/headers")
+    req = { headers, cookies }
+    res = { getHeader() {}, setCookie() {}, setHeader() {} }
+  } else {
+    req = args[0]
+    res = args[1]
+    options = args[2]
+  }
 
   options.secret = options.secret ?? process.env.NEXTAUTH_SECRET
 

--- a/packages/next-auth/src/next/index.ts
+++ b/packages/next-auth/src/next/index.ts
@@ -125,7 +125,12 @@ export async function unstable_getServerSession(
     options = args[0]
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const { headers, cookies } = require("next/headers")
-    req = { headers, cookies }
+    req = {
+      headers,
+      cookies: cookies()
+        .getAll()
+        .reduce((acc, c) => ({ ...acc, [c.name]: c.value }), {}),
+    }
     res = { getHeader() {}, setCookie() {}, setHeader() {} }
   } else {
     req = args[0]

--- a/packages/next-auth/src/next/index.ts
+++ b/packages/next-auth/src/next/index.ts
@@ -127,9 +127,11 @@ export async function unstable_getServerSession(
     const { headers, cookies } = require("next/headers")
     req = {
       headers,
-      cookies: cookies()
-        .getAll()
-        .reduce((acc, c) => ({ ...acc, [c.name]: c.value }), {}),
+      cookies: Object.fromEntries(
+        cookies()
+          .getAll()
+          .map((c) => [c.name, c.value])
+      ),
     }
     res = { getHeader() {}, setCookie() {}, setHeader() {} }
   } else {

